lib/searchkick/query.rb

```diff
@@ -1138,6 +1133,17 @@ def location_value(value)
       end
     end
 
+    def expand_range(range)
+      expanded = {}
+      expanded[:gte] = range.begin if range.begin
+
+      if range.end && !(range.end.respond_to?(:infinite?) && range.end.infinite?)
+        expanded[range.exclude_end? ? :lt : :lte] = range.end
+      end
+
+      expanded
```

Comment:
I considered using `Object#tap` here but after benchmarking it, decided it wasn't worth the (albeit small) performance hit.

```rb
require "benchmark/ips"

def explicit_return
  hash = {}

  hash[:some_value] = "some_string"

  hash
end

def tap_implicit_return
  {}.tap do |hash|
    hash[:some_value] = "some_string"
  end
end

Benchmark.ips do |b|
  b.report("explicit return") { explicit_return }
  b.report("tap implicit return") { tap_implicit_return }

  b.compare!
end
```

```
Warming up --------------------------------------
     explicit return     1.042M i/100ms
 tap implicit return   816.126k i/100ms
Calculating -------------------------------------
     explicit return     10.519M (± 0.9%) i/s -     53.152M in   5.053522s
 tap implicit return      8.177M (± 0.9%) i/s -     41.622M in   5.090811s

Comparison:
     explicit return: 10518686.7 i/s
 tap implicit return:  8176709.5 i/s - 1.29x  (± 0.00) slower
```